src/java/org/apache/cassandra/db/DeletionTime.java

```diff
@@ -187,8 +255,31 @@ public void skip(DataInputPlus in) throws IOException
 
         public long serializedSize(DeletionTime delTime)
         {
-            return TypeSizes.sizeof(delTime.localDeletionTime())
-                 + TypeSizes.sizeof(delTime.markedForDeleteAt());
+            return TypeSizes.sizeof(Integer.MAX_VALUE)
+                   + TypeSizes.sizeof(delTime.markedForDeleteAt());
+        }
+    }
+
+    // When scrubbing legacy sstables (overflown) or upon sstable corruption we could have negative ldts
+    public static class InvalidDeletionTime extends DeletionTime
+    {
+        private InvalidDeletionTime(long markedForDeleteAt, int localDeletionTime)
```

Comment:
After the last changes `localDeletionTime` isn't used anymore, since we use the generic sentinel value. So we can remove the argument.

Reply:
That I like a lot and it stresses the fact we shed the actual value and use a sentinel instead.

Reply:
Great!